products/logs/backend/api.py

```diff
@@ -24,15 +29,112 @@ def query(self, request: Request, *args, **kwargs) -> Response:
         if query_data is None:
             return Response({"error": "No query provided"}, status=status.HTTP_400_BAD_REQUEST)
 
-        query = self.get_model(query_data, LogsQuery)
+        query = LogsQuery(
+            dateRange=self.get_model(query_data.get("dateRange"), DateRange),
+            severityLevels=query_data.get("severityLevels", []),
+            orderBy=query_data.get("orderBy"),
+            searchTerm=query_data.get("searchTerm", None),
+            filterGroup=query_data.get("filterGroup", None),
+        )
         runner = LogsQueryRunner(query, self.team)
-
         try:
             response = runner.run(ExecutionMode.CALCULATE_BLOCKING_ALWAYS)
         except Exception as e:
             capture_exception(e)
             return Response({"error": "Something went wrong"}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+        assert isinstance(response, LogsQueryResponse | CachedLogsQueryResponse)
+        return Response({"query": query, "results": response.results}, status=200)
+
+    @action(detail=False, methods=["POST"], required_scopes=["error_tracking:read"])
```

Comment:
nothing too important but I guess we should change the `required_scopes` at some point to use `logs` rather than `error_tracking`